frontend/src/components/forms/Signup.tsx

```diff
@@ -32,38 +43,98 @@ const Signup = ({
   toggle,
   setToggle,
 }: SignupProps): React.ReactElement => {
+  const [signupClicked, setSignupClicked] = useState<boolean>(false);
+  const [emailError, setEmailError] = useState<boolean>(false);
+  const [emailErrorStr, setEmailErrorStr] = useState<string>("");
+  const [passwordError, setPasswordError] = useState<boolean>(false);
+  const [passwordErrorStr, setPasswordErrorStr] = useState<string>("");
+
   const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
   const history = useHistory();
 
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = e.target.value as string;
+    if (signupClicked) {
+      if (emailRegex.test(inputValue)) {
+        setEmailErrorStr("")
+        setEmailError(false)
+      } else {
+        setEmailErrorStr("Please enter a valid email.")
+        setEmailError(true)
+      }
+    }
+    setEmail(inputValue)
+  };
+
+  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = e.target.value as string;
+    setPassword(inputValue)
+
+    if (signupClicked) {
+      if (inputValue.length >= 6) {
+        setPasswordErrorStr("")
+        setPasswordError(false)
+      }
+      else {
+        setPasswordErrorStr("Password must be 6 characters long.")
+        setPasswordError(true)
+      }
+    }
+  }
+
   const onSignupClick = async () => {
+    setSignupClicked(true)
+
+    if (!emailRegex.test(email)) {
+      setEmailErrorStr("Please enter a valid email.")
+      setEmailError(true)
+      return
+    }
+
+    if (password.length < 6) {
+      setPasswordErrorStr("Password must be 6 characters long.")
+      setPasswordError(true)
+      return
+    }
+    
     const isInvited = await commonApiClient.isUserInvited(email);
-    if (isInvited) {
-      const registerResponse = await authAPIClient.register(
-        firstName,
-        lastName,
-        email,
-        password,
-      );
-      if (registerResponse) {
-        const { requiresTwoFa, authUser } = registerResponse;
-        if (requiresTwoFa) {
-          setToggle(!toggle);
-        } else {
-          localStorage.setItem(
-            AUTHENTICATED_USER_KEY,
-            JSON.stringify(authUser),
-          );
-          setAuthenticatedUser(authUser);
+    if (isInvited !== "Not Invited") {
```

Comment:
The above check would trigger as long as some string was being returned here (which this function always did, even in the error case) hence we just changed the logic around

Reply:
Could you explain what you mean by this? Do you mean that it would show not invited even if an invalid string was passed in?

Reply:
Basically, before the condition was simply `if (isInvited)`. However, the function call was returning (even in the error case) the string `Not Invited`. This meant that, even when we were _supposed_ to be erroring out, we were interpreting the API call as successful since `if (isInvited)` would be true if the string was not empty